tests/conftest.py

```diff
@@ -7092,6 +7092,111 @@ def teardown():
+        """
+        Args:
+            namespace:
+
+        Returns:
+            lists: objects of cnv workload class with default comp and aggressive compression
+
+        """
+        vm_list_agg_compr = []
+        vm_list_default_compr = []
+
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+        log.info("Setting up csi-kms-connection-details configmap")
+        kms = pv_encryption_kms_setup_factory(kv_version="v2")
+        log.info("csi-kms-connection-details setup successful")
+
+        # Create an encryption enabled storageclass for RBD
+        sc_obj_def_compr = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=kms.kmsid,
+            new_rbd_pool=True,
+        )
+
+        sc_obj_aggressive = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=kms.kmsid,
+            compression="aggressive",
+            new_rbd_pool=True,
+        )
+        vm_configs = [
+            {
+                "volume_interface": constants.VM_VOLUME_PVC,
+                "access_mode": constants.ACCESS_MODE_RWX,
+                "sc_name": sc_obj_def_compr.name,
+            },
+            {
+                "volume_interface": "PVC",
+                "access_mode": constants.ACCESS_MODE_RWX,
+                "sc_name": sc_obj_aggressive.name,
+            },
+            {
+                "volume_interface": "PVC",
+                "access_mode": constants.ACCESS_MODE_RWO,
+                "sc_name": sc_obj_def_compr.name,
+            },
+            {
+                "volume_interface": "DVT",
+                "access_mode": constants.ACCESS_MODE_RWX,
+                "sc_name": sc_obj_def_compr.name,
+            },
+            {
+                "volume_interface": "DVT",
+                "access_mode": constants.ACCESS_MODE_RWX,
+                "sc_name": sc_obj_aggressive.name,
+            },
+        ]
+
+        # Create ceph-csi-kms-token in the tenant namespace
+        kms.vault_path_token = kms.generate_vault_token()
+        kms.create_vault_csi_kms_token(namespace=namespace)
+
+        for sc_obj in [sc_obj_def_compr, sc_obj_aggressive]:
+            pvk_obj = PVKeyrotation(sc_obj)
+            pvk_obj.annotate_storageclass_key_rotation(schedule="*/3 * * * *")
+
+        # Loop through vm_configs and create the VMs using the cnv_workload fixture
+        for config in vm_configs:
+            vm_obj = cnv_workload(
+                volume_interface=config["volume_interface"],
+                access_mode=config["access_mode"],
+                storageclass=config["sc_name"],
+                pvc_size="30Gi",  # Assuming pvc_size is fixed for all
+                source_url=constants.CNV_FEDORA_SOURCE,  # Assuming source_url is the same for all VMs
+                namespace=namespace,
+            )
+            vm_obj = vm_obj[-1]
```

Comment:
what is the need for last vm_obj in the cnv_vm_objs list?

Reply:
Actually cnv_workload gives list and everytime new obj is append on vm-obj list..So I am picking up last added vm object.

Reply:
can get the index of current iteration by enumerating

Reply:
changed as per requested.